viewing_party/party.py

```diff
@@ -1,23 +1,181 @@
-    pass
+def create_movie(title, genre, rating): 
+    # if any of the vars are None, return None
+    if not title or not genre or not rating:
+        return None
+    
+    # create new dict
+    new_movie = {
+        "title": title,
+        "genre": genre,
+        "rating":rating,
+    }
+
+    return new_movie
+
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, title):
+    # Parse through the movies in watchlist
+    for movie in user_data["watchlist"]:
+
+        # if title == the movie title then operate on it
+        if title == movie["title"]:
+            user_data["watched"].append(movie)
+            user_data["watchlist"].remove(movie)
+
+    # return modified user_data
+    return user_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
 
+def get_watched_avg_rating(user_data):
+    # deal w/ empty list of watched data
+    if user_data["watched"] == []:
+        average_rating = 0.0
+        return average_rating
+
+    # creats list of all ratings
+    ratings = []
+    for movie in user_data["watched"]:
+        ratings.append(movie['rating'])
+        
+    # calculates average
+    average_rating = mean(ratings)
+    return average_rating
+
+
+def get_most_watched_genre(user_data):
+    # handles empty watchlist
+    if user_data["watched"] == []:
+        return None
+    
+    genres = []
+    # iterate through user-data to fine movie value for watched
+    for movie in user_data["watched"]:
+        # add value of genre to variable genres
+        genres.append(movie["genre"])
+        
+    # use count to find which genres repeat most and delete others
+    popular_genre = max(set(genres), key=genres.count)
+    
+    return popular_genre
+
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+
+def get_friends_watched_titles(user_data):
+    return set(movie["title"] for friend in user_data["friends"] for movie in friend["watched"])
+
+def get_user_watched_titles(user_data):
+    return set([movie["title"] for movie in user_data['watched']])
+
+def get_unique_watched(user_data):
+    # get all the titles for users_watched
+    user_watched_titles = get_user_watched_titles(user_data)
+
+    # build set of all friends watched
+    friends_watched_titles = get_friends_watched_titles(user_data)
+
+    # find unique user watched
+    unique_user_watched_titles = user_watched_titles - friends_watched_titles
+    unique_user_watched_movies = []
+    for movie in user_data["watched"]:
+        if movie["title"] in unique_user_watched_titles:
+            unique_user_watched_movies.append(movie)
+
+    return unique_user_watched_movies    

```

Comment:
👍  Yay helper functions!! Great use of set differences as well. `unique_user_watched_movies` can also be created using list comprehension.